### ISSUE TITLE
Web server not starting

### ISSUE THREAD

**Issue by 0xCUB3** [jareer12/dynamo#1]
I hate to be that guy again and ruin your Sunday even more, but running this on Ubuntu 22.04 LTS doesn't start the web server.

There are some errors in the logs but I'm not sure if they are related:
```
sudo wget -O - https://raw.githubusercontent.com/jareer12/dynamo/main/install-ubuntu.sh | sudo bash

Redirecting output to ‘wget-log.1’.
rm: cannot remove 'Dynamo': No such file or directory
rm: cannot remove 'dynamo': No such file or directory
mkdir: cannot create directory ‘/root/dynamo/static/dist’: No such file or directory
Cloning into 'Dynamo'...
remote: Enumerating objects: 2481, done.
remote: Counting objects: 100% (152/152), done.
remote: Compressing objects: 100% (95/95), done.
remote: Total 2481 (delta 82), reused 99 (delta 53), pack-reused 2329
Receiving objects: 100% (2481/2481), 10.22 MiB | 13.69 MiB/s, done.
Resolving deltas: 100% (854/854), done.

added 200 packages, and audited 201 packages in 31s

30 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> dynamo-client@1.0.0 build
> vite build

vite v2.9.12 building for production...
✓ 135 modules transformed.
dist/index.html                  0.45 KiB
dist/assets/index.394e9c6b.css   51.29 KiB / gzip: 7.37 KiB
dist/assets/index.c8fe64af.js    198.92 KiB / gzip: 69.42 KiB
assets	index.html

added 185 packages, and audited 186 packages in 8s

17 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies

changed 298 packages, and audited 299 packages in 26s

45 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (1 moderate, 6 high, 1 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

up to date, audited 186 packages in 977ms

17 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> dynamo-server@1.0.0 build
> npx tsc

warn:    --minUptime not set. Defaulting to: 1000ms
warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
info:    Forever processing file: ./src/Index.js
(node:4670) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:4670) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
nginx is already the newest version (1.18.0-6ubuntu14.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
rm: cannot remove '/etc/nginx/sites-enabled/default.save': No such file or directory
rm: cannot remove '/etc/nginx/sites-enabled/default': No such file or directory
```

**Comment by jareer12** [comment]
> I hate to be that guy again and ruin your Sunday even more, but running this on Ubuntu 22.04 LTS doesn't start the web server.
> 
> There are some errors in the logs but I'm not sure if they are related:
> 
> ```
> sudo wget -O - https://raw.githubusercontent.com/jareer12/dynamo/main/install-ubuntu.sh | sudo bash
> 
> Redirecting output to ‘wget-log.1’.
> rm: cannot remove 'Dynamo': No such file or directory
> rm: cannot remove 'dynamo': No such file or directory
> mkdir: cannot create directory ‘/root/dynamo/static/dist’: No such file or directory
> Cloning into 'Dynamo'...
> remote: Enumerating objects: 2481, done.
> remote: Counting objects: 100% (152/152), done.
> remote: Compressing objects: 100% (95/95), done.
> remote: Total 2481 (delta 82), reused 99 (delta 53), pack-reused 2329
> Receiving objects: 100% (2481/2481), 10.22 MiB | 13.69 MiB/s, done.
> Resolving deltas: 100% (854/854), done.
> 
> added 200 packages, and audited 201 packages in 31s
> 
> 30 packages are looking for funding
>   run `npm fund` for details
> 
> found 0 vulnerabilities
> 
> > dynamo-client@1.0.0 build
> > vite build
> 
> vite v2.9.12 building for production...
> ✓ 135 modules transformed.
> dist/index.html                  0.45 KiB
> dist/assets/index.394e9c6b.css   51.29 KiB / gzip: 7.37 KiB
> dist/assets/index.c8fe64af.js    198.92 KiB / gzip: 69.42 KiB
> assets	index.html
> 
> added 185 packages, and audited 186 packages in 8s
> 
> 17 packages are looking for funding
>   run `npm fund` for details
> 
> found 0 vulnerabilities
> npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
> npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
> npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
> npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
> 
> changed 298 packages, and audited 299 packages in 26s
> 
> 45 packages are looking for funding
>   run `npm fund` for details
> 
> 8 vulnerabilities (1 moderate, 6 high, 1 critical)
> 
> To address all issues (including breaking changes), run:
>   npm audit fix --force
> 
> Run `npm audit` for details.
> 
> up to date, audited 186 packages in 977ms
> 
> 17 packages are looking for funding
>   run `npm fund` for details
> 
> found 0 vulnerabilities
> 
> > dynamo-server@1.0.0 build
> > npx tsc
> 
> warn:    --minUptime not set. Defaulting to: 1000ms
> warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
> info:    Forever processing file: ./src/Index.js
> (node:4670) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:4670) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> nginx is already the newest version (1.18.0-6ubuntu14.1).
> 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
> rm: cannot remove '/etc/nginx/sites-enabled/default.save': No such file or directory
> rm: cannot remove '/etc/nginx/sites-enabled/default': No such file or directory
> ```

These aren't errors just warnings, it's the shell script installing and setting up the server. Once the script is done you can find the panel at `127.0.0.1:8085` or `127.0.0.1:8080`.

**Comment by 0xCUB3** [comment]
@jareer12 that's the thing - the panel doesn't show up and I can't curl the page either.

**Comment by jareer12** [comment]
> @jareer12 that's the thing - the panel doesn't show up and I can't curl the page either.

The files should be at /root/dynamo/static/dist. The script configures the Nginx default config file to render the frontend on port 2278, check if the config exists in your default Nginx config file.

**Comment by 0xCUB3** [comment]
@jareer12 the config exists and has the correct values. An update though - if I go to port 8085 it can't connect, but on port 8080 I get the following:
![Screenshot 2022-06-26 at 9 39 16 AM](https://user-images.githubusercontent.com/94565160/175816971-fb917100-74e1-4fa2-b058-c16da3a4dc13.png)
it's definitely an nginx error.

My /etc/nginx/sites-enabled/default:
`server { listen 8080; root /var/www/dynamo; index index.html index.htm; location / { root /var/www/dynamo; try_files $uri /index.html; } error_log /var/log/nginx/dynamo-err.log; access_log /var/log/nginx/dynamo-access.log; }`

**Comment by 0xCUB3** [comment]
this is on Ubuntu 22.04 LTS fyi in case I haven't mentioned

**Comment by jareer12** [comment]
Change `root /var/www/dynamo;` with `root /root/dynamo/static/dist;`

**Comment by jareer12** [comment]
Will fix this bug soon.

**Comment by 0xCUB3** [comment]
ok so should I edit the script with those changes and try again?

**Comment by 0xCUB3** [comment]
this is what the nginx config looks like now. Is there any easy way of restarting the server?

`server { listen 8080; root /root/dynamo/static/dist; index index.html index.htm; location / { root /root/dynamo/static/dist; try_files $uri /index.html; } error_log /var/log/nginx/dynamo-err.log; access_log /var/log/nginx/dynamo-access.log; }`

**Comment by 0xCUB3** [comment]
@jareer12 even with the above changes and the restart, I still get the 500 error when visiting http://127.0.0.1:8080 and cannot connect to :8085. lmk if I'm still doing something wrong

**Comment by jareer12** [comment]
> @jareer12 even with the above changes and the restart, I still get the 500 error when visiting http://127.0.0.1:8080 and cannot connect to :8085. lmk if I'm still doing something wrong

Is Nginx even running?

**Comment by 0xCUB3** [comment]
looks like it:
```
systemctl status nginx

● nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset:>
     Active: active (running) since Sun 2022-06-26 13:52:13 UTC; 37min ago
       Docs: man:nginx(8)
    Process: 771 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_proces>
    Process: 867 ExecStart=/usr/sbin/nginx -g daemon on; master_process on; (co>
   Main PID: 868 (nginx)
      Tasks: 5 (limit: 4474)
     Memory: 10.8M
        CPU: 73ms
     CGroup: /system.slice/nginx.service
             ├─868 "nginx: master process /usr/sbin/nginx -g daemon on; master_>
             ├─869 "nginx: worker process" "" "" "" "" "" "" "" "" "" "" "" "" >
             ├─870 "nginx: worker process" "" "" "" "" "" "" "" "" "" "" "" "" >
             ├─871 "nginx: worker process" "" "" "" "" "" "" "" "" "" "" "" "" >
             └─872 "nginx: worker process" "" "" "" "" "" "" "" "" "" "" "" "" >

Jun 26 13:52:00 raspberrypi systemd[1]: Starting A high performance web server >
Jun 26 13:52:13 raspberrypi systemd[1]: Started A high performance web server a>
lines 1-19/19 (END)
```

**Comment by 0xCUB3** [comment]
@jareer12 according to systemctl it's running 🤔 and the logs show that it started a web server, not sure if that means anything tho

**Comment by jareer12** [comment]
> @jareer12 according to systemctl it's running 🤔 and the logs show that it started a web server, not sure if that means anything tho

Hmmm, weird. I have no idea why it's not working. You'll need to provide more info or look into the problem carefully.

**Comment by 0xCUB3** [comment]
yeah it's odd. do you have any more suggestions before I venture off on my own to troubleshoot? I know that this project is in early development, which is why I'm willing to help determine what is the cause of these issues.

**Comment by jareer12** [comment]
> yeah it's odd. do you have any more suggestions before I venture off on my own to troubleshoot? I know that this project is in early development, which is why I'm willing to help determine what is the cause of these issues.

I get it now, the panel shows when the backend server starts(located at `127.0.0.1:7000`). If the backend has started you'll find the dashboard Ui at `127.0.0.1:8085`. Also double check your `.env`.

**Comment by 0xCUB3** [comment]
> > yeah it's odd. do you have any more suggestions before I venture off on my own to troubleshoot? I know that this project is in early development, which is why I'm willing to help determine what is the cause of these issues.
> 
> I get it now, the panel shows when the backend server starts(located at `127.0.0.1:7000`). If the backend has started you'll find the dashboard Ui at `127.0.0.1:8085`. Also double check your `.env`.

how will I know if the backend server started?

**Comment by 0xCUB3** [comment]
@jareer12 I'm going to install a fresh Ubuntu 22.04 LTS, I feel like some other things I did with the machine prior to this might have messed it up - idk. I'll touch base later

**Comment by 0xCUB3** [comment]
@jareer12 sadly it still doesn't work, even after a fresh install of Ubuntu 22.04 LTS...logs are the same still:
```
sudo wget -O - https://raw.githubusercontent.com/jareer12/dynamo/main/install-ubuntu.sh | sudo bash

Redirecting output to ‘wget-log.3’.
mkdir: cannot create directory ‘/root/dynamo/static/dist’: No such file or directory
Cloning into 'Dynamo'...
remote: Enumerating objects: 2481, done.
remote: Counting objects: 100% (152/152), done.
remote: Compressing objects: 100% (95/95), done.
remote: Total 2481 (delta 82), reused 99 (delta 53), pack-reused 2329
Receiving objects: 100% (2481/2481), 10.22 MiB | 1.83 MiB/s, done.
Resolving deltas: 100% (854/854), done.

added 200 packages, and audited 201 packages in 6s

30 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> dynamo-client@1.0.0 build
> vite build

vite v2.9.12 building for production...
✓ 135 modules transformed.
dist/index.html                  0.45 KiB
dist/assets/index.394e9c6b.css   51.29 KiB / gzip: 7.37 KiB
dist/assets/index.c8fe64af.js    198.92 KiB / gzip: 69.42 KiB
assets	index.html

added 185 packages, and audited 186 packages in 3s

17 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies

changed 298 packages, and audited 299 packages in 5s

45 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (1 moderate, 6 high, 1 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

up to date, audited 186 packages in 950ms

17 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> dynamo-server@1.0.0 build
> npx tsc

warn:    --minUptime not set. Defaulting to: 1000ms
warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
info:    Forever processing file: ./src/Index.js
(node:6832) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:6832) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
nginx is already the newest version (1.18.0-6ubuntu14.1).
0 upgraded, 0 newly installed, 0 to remove and 13 not upgraded.
rm: cannot remove '/etc/nginx/sites-enabled/default.save': No such file or directory
```

**Comment by jareer12** [comment]
> how will I know if the backend server started?

`forever list`
you can manually restart by `cd /root/dynamo && node .`

**Comment by anastarawneh** [comment]
I'm having the same issue here, on the same Ubuntu version. I edited the nginx `default` file, and when I run `cd /root/dynamo && node .` the output is this:
```
MySQL config not found
MySQL has not connected yet
MySQL has not connected yet
MySQL has not connected yet
MySQL has not connected yet
MySQL has not connected yet
server { listen 8080; root /root/dynamo/static/dist; index index.html index.htm; location / { root /root/dynamo/static/dist; try_files $uri /index.html; } error_log /var/log/nginx/dynamo-err.log; access_log /var/log/nginx/dynamo-access.log; }

MySQL has not connected yet
Dashboard UI Available on http://127.0.0.1:undefined
Backend Server Instance Available at http://127.0.0.1:undefined
```
Port 8080 still returns a 500 Internal Server Error.

I also tried reinstalling with the original installation command and running `node .` again, and it gives me the same response, and still returns 500.

**Comment by jareer12** [comment]
> EDIT: I think this might be it: I have no idea where the .env file is, and I don't think it was installed properly.

You're right the `.env` didn't install properly.
